base/gmp.jl

```diff
@@ -41,13 +41,30 @@ type BigInt <: Integer
     size::Cint
     d::Ptr{Limb}
     function BigInt()
-        b = new(zero(Cint), zero(Cint), C_NULL)
-        ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
-        finalizer(b, _gmp_clear_func)
+        b::BigInt
+        if length(bigintpool) > 0
+            b = pop!(bigintpool)
+            x= string(b) #########   <------- segfaulting!
```

Comment:
`b` is dead here. This line raises a segfault at `__gmpz_get_str`. But `b` was not dead when inserted in the pool during finalization. So I think it was killed by the GC after finalization. Can this be?